### PR TITLE
[SYCLomatic] Enable migration of cudaStreamCaptureStatus enum and SYCL graphs exp option

### PR DESCRIPTION
**Pull request by daiyaan-ahmed6**
Signed-off-by: Daiyaan Ahmed <daiyaan.ahmed@intel.com>

### CODE REVIEW

**Review by tomflinda — APPROVED** [review]
LGTM

File: clang/lib/DPCT/ASTTraversal.cpp [hunk 1]
```diff
@@ -3398,6 +3400,9 @@ void EnumConstantRule::runRule(const MatchFinder::MatchResult &Result) {
       EnumName == "cudaComputeModeExclusiveProcess") {
     handleComputeMode(EnumName, E);
     return;
+  } else if (EnumName == "cudaStreamCaptureStatusInvalidated") {
+    report(E->getBeginLoc(), Diagnostics::KNOWN_UNSUPPORTED_TYPE, true,
```
Comment (tangjj11) [inline clang/lib/DPCT/ASTTraversal.cpp]: It is not a type, you need report error like this:
```c++
// CHECK:     /*
// CHECK:     DPCT1007:{{[0-9]+}}: Migration of CUDNN_BATCHNORM_SPATIAL_PERSISTENT is not supported.
// CHECK:     */
// CHECK:     dpct::dnnl::batch_normalization_mode m = dpct::dnnl::batch_normalization_mode::spatial;
    cudnnBatchNormMode_t m = CUDNN_BATCHNORM_SPATIAL_PERSISTENT;
```

**Review by tangjj11 — APPROVED** [review]

**Review by ShengchenJ — APPROVED** [review]

File: clang/lib/DPCT/ASTTraversal.cpp (hunk 1 above)
Comment (daiyaan-ahmed6) [reply clang/lib/DPCT/ASTTraversal.cpp]: Done.

File: clang/include/clang/DPCT/DPCTOptions.inc [hunk 2]
```diff
@@ -758,6 +758,10 @@ DPCT_ENUM_OPTION(
             "bindless_images", int(ExperimentalFeatures::Exp_BindlessImages),
             "Experimental extension that allows use of bindless images APIs.\n",
             false),
+        DPCT_OPTION_ENUM_VALUE(
+            "graphs", int(ExperimentalFeatures::Exp_Graphs),
```
Comment (zhimingwang36) [inline clang/include/clang/DPCT/DPCTOptions.inc]:
```suggestion
            "graph", int(ExperimentalFeatures::Exp_Graphs),
```

File: clang/test/dpct/cuda-stream-api-cuda10-after.cu [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  // CHECK-NEXT: captureStatus = cudaStreamCaptureStatusInvalidated;
+  cudaStreamCaptureStatus captureStatus = cudaStreamCaptureStatusNone;
+  captureStatus = cudaStreamCaptureStatusActive;
+  captureStatus = cudaStreamCaptureStatusInvalidated;
+
+  // CHECK: /*
+  // CHECK-NEXT: DPCT1027:{{[0-9]+}}: The call to cudaStreamIsCapturing was replaced with 0 because SYCL currently does not support capture operations on queues.
+  // CHECK-NEXT: */
+  // CHECK: MY_ERROR_CHECKER(0);
+  MY_ERROR_CHECKER(cudaStreamIsCapturing(s0, &captureStatus));
```
Comment (zhimingwang36) [inline clang/test/dpct/cuda-stream-api-cuda10-after.cu:32]: from  queue_state,  can it know the a queue is in Capturing status?

**Pull request closed without merging**